test/integration/e2e/index.js

```diff
@@ -277,7 +273,7 @@ describe('basic E2E tests', () => {
                 .setPubKeyIdFromPrivateKey(alicePrivateKey).sign(alicePrivateKey);
 
             let inputs = await dapiClient.getUTXO(faucetAddress);
-            expect(inputs.items).to.have.lengthOf(2);
+            expect(inputs.items).to.have.lengthOf(1);
 
             const transaction = Transaction()
                 .setType(Transaction.TYPES.TRANSACTION_SUBTX_REGISTER)
```

Comment:
I'm wondering why inputs.items should always have just 2 elements? Doesn't this depend on how many txs have been made to the faucetAddress and could be anything between 0 and 999?

Reply:
good question. indeed in these tests size of inputs.items equals 1 and doesn't make sense to use the latest input as: inputs.items.slice(-1)[0]. I'll add separate tests that will use inputs as array with size >1 and < 999

Reply:
@Cofresi I've added new tests to verify how many inputs we could pass to Transaction https://github.com/dashevo/dapi-client/pull/53/commits/46eb4e501e91bd29b4754698b976188de63f50a3

1) with last 15 inputs( inputs.items.slice(-15)) transaction is rejected with 'rate limited free transaction'
2) with last 15 inputs everything is fine

Do you think this is a valid behavior?

Reply:
I think there is something wrong with this. Still investigating, but this is where its coming from:
https://github.com/dashpay/dash/blob/master/src/validation.cpp#L841
It means with 15 inputs the size of the transaction needs a bigger fee to pass this validation test. So I guess what decides that the test errors with this message is not only the mere number of 15 inputs, but also how much decide to pay the miners. In this specific tx, even if you add a lot of fees as credits to the OP_RETURN output of the TRANSACTION_SUBTX_REGISTER, this still doesn't count to the fees you are paying the miner. The classical tx fees are calculated as 
cf: https://github.com/dashpay/dips/blob/master/dip-0005.md#subscription-transaction-fees
My hypothesis right now is that adding the credit output doesn't actually count as an output sum and that all of the available input amount is added to the change output which would leave nothing to the miner.
Will investigate further...